.github/workflows/cromwell_unit_tests.yml

```diff
@@ -0,0 +1,86 @@
+name: 'Cromwell unit tests'
+
+#This github action runs all of Cromwell's unit tests.
+#A matrix strategy is used so that each subproject can be unit tested in parallel with all other subprojects. 
+
+#This is what shows up in the github workflows page as the title. 
+run-name: ${{ github.actor }} running Cromwell sbt unit tests.
+
+#What will trigger the workflow to run. 
+on:
+  workflow_dispatch: #Manual trigger from GitHub UI
+  push:
+  pull_request: 
+
+permissions: 
+  contents: read
+
+jobs:
+  build-and-test: 
+    strategy:
+      fail-fast: false #Disabling fail fast means that even if one job in the matrix fails, the others will still try to complete.
+      matrix: #this array is the output of the command `sbt projects` from the cromwell root directory (minus lots of formatting)
+        sbtSubProject: [awsBackend,
+        awsS3FileSystem,
+        azureBlobFileSystem,
+        backend,
+        centaur,
+        cloud-nio-impl-drs,
+        cloud-nio-impl-ftp,
+        cloud-nio-spi,
+        cloud-nio-util,
+        cloudSupport,
+        common,
+        core,
+        cromiam,
+        cromwell-drs-localizer,
+        cromwellApiClient,
+        databaseMigration,
+        databaseSql,
+        dockerHashing,
+        drsFileSystem,
+        engine,
+        ftpFileSystem,
+        gcsFileSystem,
+        googlePipelinesCommon,
+        googlePipelinesV2Alpha1,
+        googlePipelinesV2Beta,
+        httpFileSystem,
+        languageFactoryCore,
+        perf,
+        server,
+        services,
+        sfsBackend,
+        sraFileSystem,
+        tesBackend,
+        wdlBiscayneLanguageFactory,
+        wdlDraft2LanguageFactory,
+        wdlDraft3LanguageFactory,
+        wdlModelBiscayne,
+        wdlModelDraft2,
+        wdlModelDraft3,
+        wdlNewBaseTransforms,
+        wdlSharedModel,
+        wdlSharedTransforms,
+        wdlTransformsBiscayne,
+        wdlTransformsDraft2,
+        wdlTransformsDraft3,
+        wes2cromwell,
+        wom,
+        womtool]
+
+    #This action is using a Github free runner, rather than a Broad self-hosted one.
+    #This is because the Broad ones don't have sbt installed by default.
+    runs-on: ubuntu-latest
```

Comment:
That's probably fine for these small tests, that are unlikely to ever be the longest-running piece and therefore gating factor to speed.